src/typechecker/FStar.TypeChecker.Rel.fs

```diff
@@ -3847,6 +3876,55 @@ let teq_nosmt (env:env) (t1:typ) (t2:typ) : option<guard_t> =
   | None -> None
   | Some g -> discharge_guard' None env g false
 
+(*
+ * Solve the uni-valued implicits
+ *
+ * For now we handle only unit and unit refinement typed implicits,
+ *   we can later extend it to single constructor inductives
+ *
+ * This function gets the unresolved implicits from the main resolve_implicits'
+ *   function
+ *
+ * It only sets the value of the implicit's ctx uvar in the UF graph
+ * -- leaving their typechecking to resolve_implicits'
+ *
+ * E.g. for a ?u:squash phi, this will only set ?u=unit in the UF graph,
+ *   and, as usual, resolve_implicits' will check that G |= phi
+ *
+ * It returns a boolean (true if at least one implicit was solved)
+ *   and the set of new implicits, right now this set is same as imps,
+ *   for inductives, this may later include implicits for pattern variables
+ *)
+ 
```

Comment:
This part is from PR #2041, right? Can we merge that separately?

Reply:
It would be tricky, I have made some changes to this code ... e.g. not having this code kick in for tactics. If you have some comments on that PR, I will be happy to accommodate them here, but it would be easier for me to merge it as part of this PR rather than porting changes to that branch and then do it